veb.c:207: error: 'T.M' may be used uninitialized in this function

On current master, with default gcc from TOSS 2 system (hype) 
```
$ gcc --version
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-17)
```

flux-core can't be built due to the new error 

```
cc1: warnings being treated as errors
veb.c: In function 'vebnew':
veb.c:207: error: 'T.M' may be used uninitialized in this function
make[3]: *** [veb.lo] Error 1
make[3]: Leaving directory `/g/g0/grondo/git/f/src/common/libutil'
make[2]: *** [all-recursive] Error 1
```

I have to admit I'm not sure why the compiler is complaining about the new code vs the old version, though initializing `T.M = 0` makes the warning go away -- I don't feel confident about that to actually propose a PR though... Honestly since newer compilers don't complain it could be compiler bug.

```diff
diff --git a/src/common/libutil/veb.c b/src/common/libutil/veb.c
index 0bac5fd..9c67e67 100644
--- a/src/common/libutil/veb.c
+++ b/src/common/libutil/veb.c
@@ -205,6 +205,7 @@ Veb
 vebnew(uint M, int full)
 {
        Veb T;
+       T.M = 0;
        T.k = fls(M-1);
        T.D = malloc(vebsize(M));
        if (T.D) {

```

--- a/src/common/libutil/veb.c
+++ b/src/common/libutil/veb.c
@@ -205,6 +205,7 @@ Veb
 vebnew(uint M, int full)
 {
 	Veb T;
+	T.M = 0;
 	T.k = fls(M-1);
 	T.D = malloc(vebsize(M));
 	if (T.D) {

